model/common/src/icon4py/model/common/states/factory.py

```diff
@@ -611,13 +623,18 @@ def _compute(
         args.update(offsets)
         args.update(self._params)
         results = self._func(**args)
-        # TODO(): can the order of return values be checked?
-        results = (results,) if isinstance(results, data_alloc.NDArray) else results
+        # convert to tuple
+        results = (results,) if not isinstance(results, tuple) else results
         self._fields = {
-            k: gtx.as_field(tuple(self._dims), results[i], allocator=backend)
+            k: self._as_field(backend, results[i]) if self._dims else results[i]
             for i, k in enumerate(self.fields)
         }
 
+    def _as_field(
+        self, backend: gtx_typing.Backend | None, value: data_alloc.NDArray
+    ) -> state_utils.GTXFieldType:
+        return gtx.as_field(tuple(self._dims), value, allocator=backend)
```

Comment:
Are the RBF tests failing perhaps because I made them (at least the cell and vertex; edge is only one field) return lists instead of tuples (as the type hints would indicate)...? Or should lists of fields as return values be allowed? NB. I'm not asking for it to be allowed, I think restricting it to tuples and single arrays is sufficient.

Reply:
maybe. I dont know whether it matters at runtime. I never thougth about because we return tuples in gt4py and if you just do a 
```
def foo():
    ...
    return a, b, c
```
that will be a tuple.

Reply:
I did a tuple([...]). The (...) will return a generator expression, not a tuple...